prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -359,6 +359,31 @@ class ActionHistory {
         trackEvent(event) // to be sent to queue later.
     }
 
+    fun trackCreatedReport(
+        event: Event,
+        report: Report,
+        blobInfo: BlobAccess.BlobInfo
+    ) {
+        if (isReportAlreadyTracked(report.id)) {
+            error("Bug:  attempt to track history of a report ($report.id) we've already associated with this action")
+        }
+
+        val reportFile = ReportFile()
+        reportFile.reportId = report.id
+        reportFile.nextAction = event.eventAction.toTaskAction()
+        reportFile.nextActionAt = event.at
+        reportFile.schemaName = report.schema.name
+        reportFile.schemaTopic = report.schema.topic
+        reportFile.bodyUrl = blobInfo.blobUrl
+        reportFile.bodyFormat = blobInfo.format.toString()
+        reportFile.blobDigest = blobInfo.digest
+        reportFile.itemCount = report.itemCount
+        reportsOut[reportFile.reportId] = reportFile
+        filteredReportRows[reportFile.reportId] = report.filteredItems
+        trackItemLineages(report)
+        trackEvent(event) // to be sent to queue later.
+    }
+
```

Comment:
I'm not requesting a change, but I really like writing this kind of code using an `apply` block:

```
val reportFile = ReportFile().apply {
    reportId = report.id
    nextAction = event.eventAction.toTaskAction()
   ...
}
```

Reply:
More info here: https://kotlinlang.org/docs/scope-functions.html#context-object-this-or-it

Reply:
nice. yeah this is an existing pattern that should be refactored, either to that or to constructors of the `ReportFile` class.